### DISCUSSION TITLE
While Blazor Runtime is Wasm, Oqtane are not stable.

### DISCUSSION THREAD

**Discussion by Line4GitHub** [oqtane/oqtane.framework#1748] [General]
<html>
<body>
<p>While Blazor Runtime is Wasm, Oqtane are not stable.</p>
</body>
</html>

**Comment by sbwalker** [comment]
@Line4GitHub is it possible for you to provide some specific examples where you encountered instability? It would also be helpful to understand which version of Oqtane/.NET you are using.

**Reply by Line4GitHub** [reply]
1.  While set Blazor Runtime to Webassembly in Oqtane 2.3 (Install with Oqtane.Framework.2.3.0.Install.zip),
2. 2sxc App & Content version:12.08.00 (Install with Module Management)  there are some (ie. Swiper Slider, Person,,,) render problem.
3. Blog module for Oqtane version: 1.0.3, (Install with Module Management) after Installed & restart, The 2sxc App & Content are all encountered Error: System.AggregateException: One or more errors occurred. (Empty path name is not legal. (Parameter 'path')) ...
4. After Uninstall  Blog module for Oqtane version: 1.0.3, there (2sxc App & Content) are no Error again.

**Reply by Line4GitHub** [reply]
If ReInstall Oqtane 2.3 with Oqtane.Framework.2.3.0.Install.zip, Then Install Blog module for Oqtane version: 1.0.3 with Module Management.
Then set Blazor Runtime to Webassembly.
The blog content only 1st time not disappear, when edit.

**Reply by sbwalker** [reply]
@Line4GitHub I assume you mean Oqtane 2.3.1 - as Oqtane 2.3.0 had a fairly major bug related to AntiForgeryTokens which required the publication of a 2.3.1 release almost immediately.

It sounds like you are experiencing instability with third party modules installed in Oqtane when running on WebAssembly. For 2SXC you would need to contact the provider of the module to determine if they are aware of the issues you are experiencing. The Blog module is not part of the Oqtane framework - so it would need to be investigated separately. Are you having any issues with the Oqtane framework itself when running on WebAssembly ( ie. unrelated to third party modules )?

**Reply by Line4GitHub** [reply]
Thank you very much for your reply!
Now there is no any issues with the Oqtane framework itself when running on WebAssembly!
Thank you very much for your reply!

**Reply by sbwalker** [reply]
One of the challenges with Blazor WebAssembly is ts dependency on the linker and the “magic” which happens under the covers when you publish a release.  The linker removes classes and methods from the core (System.*) libraries that it thinks the application isn’t using. So, if the linker sees 10 assemblies in the app, and doesn’t see anyone calling `string.Replace`, the linker will remove the `Replace` method from System.String. However, if the application dynamically loads up an 11th assembly that does call `string.Replace`, the application is going to crash. However it is also important to understand why the linker removes methods - it does this to optimize the code that is downloaded to the browser to ensure the application performs well on the first load.

**Reply by Line4GitHub** [reply]
Thank you very much for your detailed explanation.
